base/compiler/abstractinterpretation.jl

```diff
@@ -1620,7 +1627,7 @@ function abstract_invoke(interp::AbstractInterpreter, (; fargs, argtypes)::ArgIn
     ti = tienv[1]; env = tienv[2]::SimpleVector
     result = abstract_call_method(interp, method, ti, env, false, sv)
     (; rt, edge, effects) = result
-    edge !== nothing && add_backedge!(edge::MethodInstance, sv)
+    edge !== nothing && add_backedge!(edge::MethodInstance, sv, argtypes)
```

Comment:
Might be helpful to avoid recomputing this?
```suggestion
    edge !== nothing && add_backedge!(edge::MethodInstance, sv, types)
```